### PR TITLE
Refine DataFrame/Series.quantile.

### PR DESCRIPTION
**Pull request by ueshin**
Refines `DataFrame/Series.quantile` to:

- Reuse `_reduce_for_stat_function` when `q` is `float`.
- Consolidate the logic when `q` is `Iterable`.

Also support `numeric_only` for `DataFrame`.

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -10120,7 +10120,11 @@ def info(self, verbose=None, buf=None, max_cols=None, null_counts=None) -> None:
 
     # TODO: fix parameter 'axis' and 'numeric_only' to work same as pandas'
     def quantile(
-        self, q=0.5, axis=0, numeric_only=True, accuracy=10000
+        self,
+        q: Union[float, Iterable[float]] = 0.5,
+        axis: Union[int, str] = 0,
```
Comment (xinrong-databricks) [inline databricks/koalas/frame.py:10125]: nit: Should `axis : **int**, default 0 or 'index'`in docstring be `int or str`?

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -3261,91 +3263,49 @@ def quantile(self, q=0.5, accuracy=10000) -> Union[Scalar, "Series"]:
...
+                raise ValueError(
+                    "accuracy must be an integer; however, got [%s]" % type(accuracy).__name__
+                )
 
-        for v in q if isinstance(q, list) else [q]:
-            if not isinstance(v, float):
+            if not isinstance(q, float):
                 raise ValueError(
-                    "q must be a float of an array of floats; however, [%s] found." % type(v)
+                    "q must be a float of an array of floats; however, [%s] found." % type(q)
```
Comment (xinrong-databricks) [inline databricks/koalas/series.py:3295]: `q must be a float **or** an array of floats`?

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -10162,86 +10166,113 @@ def quantile(
...
+                    percentile_col = quantile(self._internal.spark_column_for(label), spark_type)
+                    percentile_cols.append(percentile_col.alias(column))
+                    percentile_col_names.append(column)
+                    column_labels.append(label)
+
+            if len(percentile_cols) == 0:
+                return DataFrame(index=q)
+
+            sdf = self._internal.spark_frame.select(percentile_cols)
+            # Here, after select percntile cols, a spark_frame looks like below:
```
Comment (xinrong-databricks) [inline databricks/koalas/frame.py:10236]: nit: `percentile`?

File: databricks/koalas/frame.py [hunk 4]
```diff
@@ -10162,86 +10166,113 @@ def quantile(
...
+                for i in range(len(q)):
+                    cols_dict[column].append(scol_for(sdf, column).getItem(i).alias(column))
+
+            internal_index_column = SPARK_DEFAULT_INDEX_NAME
+            cols = []
+            for i, col in enumerate(zip(*cols_dict.values())):
+                cols.append(F.struct(F.lit(q[i]).alias(internal_index_column), *col))
+            sdf = sdf.select(F.array(*cols).alias("arrays"))
+
+            # And then, explode it and manually set the index.
```
Comment (xinrong-databricks) [inline databricks/koalas/frame.py:10255]: These figures make understanding the code so much easier 👍 .

**Comment by xinrong-databricks** [comment]
LGTM!

**Review by xinrong-databricks — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**